.github/workflows/site.yaml

```diff
@@ -59,16 +59,16 @@ jobs:
           git add .
           git commit -m 'docs/main'
       - run: |
-          wget https://github.com/apache/incubator-celeborn/archive/refs/tags/v0.2.1-incubating.tar.gz
-          tar -xzf v0.2.1-incubating.tar.gz
-          cd incubator-celeborn-0.2.1-incubating
+          wget https://github.com/apache/incubator-celeborn/archive/refs/tags/v0.3.0-incubating.tar.gz
```

Comment:
Shouldn't we retain 0.2.1 here?

Reply:
we should retain all history-versioned docs

Reply:
i think that this code should be update to  produce history docs.
![image](https://github.com/apache/incubator-celeborn-website/assets/96606293/18172552-51ae-4938-b5e3-10458315dd9a)

Reply:
This script needs to improve, e.g. extract the duplicated code. Ideally, we only need to update the version list when a new release happens.

Anyway, the optimization is not a blocker of this PR, copy-paste is fine, and we can optimize it later.